core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockStore.java

```diff
@@ -0,0 +1,233 @@
+      // Overwrites the key if it already exists.
+      mDb.put(mBlockMetaColumn, Longs.toByteArray(id), meta.toByteArray());
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void removeBlock(long id) {
+    try {
+      mDb.delete(mBlockMetaColumn, Longs.toByteArray(id));
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void clear() {
+    try {
+      initDb(mBaseDir);
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public List<BlockLocation> getLocations(long id) {
+    RocksIterator iter =
+        mDb.newIterator(mBlockLocationsColumn, new ReadOptions().setPrefixSameAsStart(true));
+    iter.seek(Longs.toByteArray(id));
+    List<BlockLocation> locations = new ArrayList<>();
+    for (; iter.isValid(); iter.next()) {
+      try {
+        locations.add(BlockLocation.parseFrom(iter.value()));
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
+    return locations;
+  }
+
+  @Override
+  public void addLocation(long id, BlockLocation location) {
+    byte[] key = toByteArray(id, location.getWorkerId());
+    try {
+      mDb.put(mBlockLocationsColumn, key, location.toByteArray());
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void removeLocation(long blockId, long workerId) {
+    byte[] key = toByteArray(blockId, workerId);
+    try {
+      mDb.delete(mBlockLocationsColumn, key);
+    } catch (RocksDBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public Iterator<Block> iterator() {
+    RocksIterator iter =
+        mDb.newIterator(mBlockMetaColumn, new ReadOptions().setPrefixSameAsStart(true));
+    iter.seekToFirst();
+    return new Iterator<Block>() {
+      @Override
+      public boolean hasNext() {
+        return iter.isValid();
+      }
+
+      @Override
+      public Block next() {
+        try {
+          return new Block(Longs.fromByteArray(iter.key()), BlockMeta.parseFrom(iter.value()));
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        } finally {
+          iter.next();
+        }
+      }
+    };
+  }
+
+  // Implementation based on
+  // https://google.github.io/guava/releases/21.0/api/docs/src-html/com/google/common/primitives/
+  // Longs.html#line.267
+  private byte[] toByteArray(long long1, long long2) {
+    byte[] key = new byte[16];
+    for (int i = 7; i >= 0; i--) {
+      key[i] = (byte) (long1 & 0xffL);
+      long1 >>= 8;
+    }
+    for (int i = 15; i >= 8; i--) {
+      key[i] = (byte) (long2 & 0xffL);
+      long2 >>= 8;
+    }
+    return key;
+  }
+
+  private void initDb(String baseDir) throws RocksDBException {
+    if (mDb != null) {
+      // Column handles must be closed before closing the db, or an exception gets thrown.
+      mDefaultColumn.close();
```

Comment:
in case some column failed to close, shall we continue? if yes, shall we use Closer?

Reply:
added error handling so that we just log an error and continue if any of these closes fails. 

Tried to use `Closer` but it is not compatible since the objects don't implement `Closeable`.